lib/saluki-event/src/lib.rs

```diff
@@ -64,48 +64,48 @@ pub enum Event {
     /// A metric.
     Metric(Metric),
 
-    /// An event.
+    /// A Datadog Event.
     EventD(EventD),
 
     /// A service check.
     ServiceCheck(ServiceCheck),
 }
 
 impl Event {
-    /// Converts this event into `Metric`.
+    /// Returns the inner event value, if this event is a `Metric`.
     ///
-    /// If the underlying event is not a `Metric`, this will return `None`.
-    pub fn into_metric(self) -> Option<Metric> {
+    /// Otherwise, `None` is returned and the original event is consumed.
+    pub fn try_into_metric(self) -> Option<Metric> {
         match self {
             Event::Metric(metric) => Some(metric),
             _ => None,
         }
     }
 
-    /// Converts this event into `Metric`.
+    /// Returns a mutable reference inner event value, if this event is a `Metric`.
     ///
-    /// If the underlying event is not a `Metric`, this will return `None`.
-    pub fn as_metric_mut(&mut self) -> Option<&mut Metric> {
+    /// Otherwise, `None` is returned.
+    pub fn try_as_metric_mut(&mut self) -> Option<&mut Metric> {
         match self {
             Event::Metric(metric) => Some(metric),
             _ => None,
         }
     }
 
-    /// Converts this event into `Eventd`.
+    /// Returns the inner event value, if this event is a `EventD`.
     ///
-    /// If the underlying event is not a `Eventd`, this will return `None`.
-    pub fn into_eventd(self) -> Option<EventD> {
+    /// Otherwise, `None` is returned and the original event is consumed.
+    pub fn try_into_eventd(self) -> Option<EventD> {
         match self {
             Event::EventD(eventd) => Some(eventd),
             _ => None,
         }
     }
```

Comment:
I updated the name of these methods because I gave you a bad suggestion in the original PR.

The common naming practice for conversion methods in Rust is:

- `into_foo` if it _consumes_ the value and the conversion isn't fallible (i.e. no `Option<T>` or `Result<T, E>`)
- `as_foo` if it provides an immutable reference conversion (i.e. `String::as_str` returns `&str`)
- `as_foo_mut` if it provides a mutable reference conversion (i.e. `Vec::as_mut_slice` returns `&mut [T]`)
- `try_into_foo`/`try_as_foo`/`try_as_foo_mut` if the conversion is fallible (i.e. how we're doing it because we don't know if an `Event` is actually a `Metric` or not)